### ISSUE TITLE
Git Credentials

### ISSUE THREAD

**Issue by tigerclawn82** [fsprojects/Paket#1804]
### Description

Git user/password configuration,  Could not run git ls-remote
Error:
bash: /dev/tty: No such device or address
error: failed to execute prompt script (exit code 1)
fatal: could not read Username for 'http://192.168.0.112': Invalid argument
### Repro steps
1. add git repository in packet.dependencies
2. .paket\paket.exe install
### Expected behavior

It should download git repository added in packet.dependencies
### Actual behavior

Giving up credentials error, normal cmd ask for username and password. Is there anyway i can give that information. credentils.xml, is that can be used?
### Related information
- Windows Server 2008 R2
- Paket version 3.6.1.0 Stable
- .NET Runtime

**Comment by forki** [comment]
we are using your default git.exe

**Comment by tigerclawn82** [comment]
Hi, thanks for your quick response. 
default git.exe ask username and password in command line but i cannot give when .paket\paket.exe install.
Is there anyway i can configure credentials? like credentials.xml?

**Comment by forki** [comment]
You can use the git credential store that can be downloaded from the git
homepage

**Comment by tigerclawn82** [comment]
Hi, i used following guide as you said use Git credential manager 

https://github.com/Microsoft/Git-Credential-Manager-for-Windows

Now, i am able to run .paket\paket.exe install

Thanks

**Comment by tigerclawn82** [comment]
Hi,
    .paket\paket.exe install is working fine locally on Build Server but its not working during the build. I am getting 
```
bash: /dev/tty: No such device or address
error: failed to execute prompt script (exit code 1)
fatal: could not read Username for 'http://192.168.0.112': Invalid argument
```
Is there anything need to be configured? I am using gitlab and is there anything i can check?

**Comment by forki** [comment]
sounds like the process doesn't retrieve the credentials from the store.

**Comment by tigerclawn82** [comment]
i have restarted the build server but still it didn't work.

**Comment by forki** [comment]
does it work "manually" from the build server?

**Comment by tigerclawn82** [comment]
yes, i checked on the build server, .paket\paket.exe install works fine

**Comment by forki** [comment]
then I assume the build runner has a user that doesn't have access to credential store

**Comment by tigerclawn82** [comment]
I have checked the gitlab-runner assigned user account and it is same as i am login with.

**Comment by tigerclawn82** [comment]
Hi,
   I have switched gitlab-runner to particular user account and now when i run paket during build, its giving error
```
Message: Could not run "git fetch -f --tags".
Error: 
fatal: Authentication failed for 'http://192.168.0.112/Shared/Infrastructure.git/'
```
When i manually clone repo in folder, it asks me password but when i again clone it, it goes without password. But once i again re-run build, it again ask for password.

It would be great if something explicit can be used for credentials.

**Comment by forki** [comment]
I'm not really sure what we can do here. We completely rely on gits own auth methods. Maybe try a different clone url which uses a different protocol.

**Comment by tigerclawn82** [comment]
gitlab build also adds 
```
remote.origin.url=http://gitlab-ci-token:k3fkQu-_z5oTzAWUFwmd@192.168.0.112/Lab/LabInterface.git
```
which is not added when i manual clone the repo

**Comment by tigerclawn82** [comment]
can i use git urls with username in it eg
```
http://wahmed@192.168.0.112
```

**Comment by tigerclawn82** [comment]
i have tried using username in git url and it doesn't help. Whenever gitlab-ci build is run, git credentials are lost.
